CMakeLists.txt

```diff
@@ -131,6 +131,37 @@ set(IREE_RUNTIME_OPTIMIZATION_PROFILE "" CACHE STRING
 set(IREE_LTO_MODE "full" CACHE STRING "LTO type, 'thin' or 'full'. Only consulted on clang-like compilers.")
 option(IREE_VISIBILITY_HIDDEN "Builds all C/C++ libraries with hidden visibility" ON)
 
+#-------------------------------------------------------------------------------
+# Custom runtime allocator configuration
+#-------------------------------------------------------------------------------
+# See iree/base/allocator.h for more information on `iree_allocator_system`.
+
+# A default allocator can be specified by C defines
+# (`IREE_ALLOCATOR_SYSTEM_CTL`) and linked in manually or by name with the
+# `IREE_ALLOCATOR_SYSTEM` option at the CMake level. If set in CMake additional
+# sources, copts, and dependencies can be added to iree::base in order to make
+# the allocator accessible.
+#
+# Built-in allocators available:
+# - `libc`: whatever `malloc`/`free` are defined as.
```

Comment:
What's compatibility like between this user-defined `IREE_ALLOCATOR_SYSTEM` and the compiler tracing memory profiling code added back in https://github.com/iree-org/iree/pull/11998 ? https://github.com/iree-org/iree/blob/55d86cd68fe84ff12614ee3fa271b77c53c1d680/compiler/src/iree/compiler/Utils/TracingUtils.cpp#L155-L242

That code currently calls `malloc` and `free` directly, but maybe it could use `iree_allocator_system` ...? Should it continue to use `malloc`/`free`, given that it overrides the system `new`/`delete`? I might be the only one who has used that code path, so we could also let it bit rot...

/me ducks and runs away